Sources/WalletConnectEcho/EchoClient.swift

```diff
@@ -3,17 +3,39 @@ import Foundation
 public class EchoClient: EchoClientProtocol {
     private let registerService: EchoRegisterService
 
+    /// The property is used to determine whether echo.walletconnect.org will be used
+    /// in case echo.walletconnect.com doesn't respond for some reason (most likely due to being blocked in the user's location).
+    private var fallback = false
+    
     init(registerService: EchoRegisterService) {
         self.registerService = registerService
     }
 
     public func register(deviceToken: Data) async throws {
-        try await registerService.register(deviceToken: deviceToken)
+        do {
+            try await registerService.register(deviceToken: deviceToken)
+        } catch {
+            if (error as? HTTPError) == .couldNotConnect && !fallback {
+                fallback = true
+                await registerService.echoHostFallback()
+                try await registerService.register(deviceToken: deviceToken)
+            }
+            throw error
```

Comment:
could we move the implementation down to registerService instead of handling it in the client?
so client stays as simple as possible.

Reply:
I think it could be even done in a separate class with a single responsibility